src/nft/components/profile/list/SetDurationModal.tsx

```diff
@@ -0,0 +1,141 @@
+  color: ${({ theme }) => theme.accentCritical};
+  gap: 4px;
+`
+
+const WarningIcon = styled(AlertTriangle)`
+  width: 16px;
+  color: ${({ theme }) => theme.accentCritical};
+`
+
+enum Duration {
+  hour = 'hour',
+  day = 'day',
+  week = 'week',
+  month = 'month',
+}
+
+enum ErrorState {
+  valid,
+  empty,
+  overMax,
+}
+
+export const SetDurationModal = () => {
+  const [duration, setDuration] = useState(Duration.day)
+  const [displayDuration, setDisplayDuration] = useState(Duration.day)
+  const [amount, setAmount] = useState(7)
+  const [errorState, setErrorState] = useState(ErrorState.valid)
+  const setGlobalExpiration = useSellAsset((state) => state.setGlobalExpiration)
+  const setCustomExpiration = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAmount(!!event.target.value.length ? parseFloat(event.target.value) : 0)
+    setDuration(displayDuration)
+  }
+  const selectDuration = (duration: Duration) => {
+    setDuration(duration)
+    setDisplayDuration(duration)
+  }
+  const durationOptions: DropDownOption[] = useMemo(
+    () => [
+      {
+        displayText: 'Hours',
+        onClick: () => selectDuration(Duration.hour),
+      },
+      {
+        displayText: 'Days',
+        onClick: () => selectDuration(Duration.day),
+      },
+      {
+        displayText: 'Weeks',
+        onClick: () => selectDuration(Duration.week),
+      },
+      {
+        displayText: 'Months',
+        onClick: () => selectDuration(Duration.month),
+      },
+    ],
+    []
+  )
+  useEffect(() => {
+    const expiration = convertDurationToExpiration(amount, duration)
+    if (expiration * 1000 - Date.now() < ms`60 seconds`) setErrorState(ErrorState.empty)
+    else if (expiration * 1000 - Date.now() > ms`180 days`) setErrorState(ErrorState.overMax)
+    else setErrorState(ErrorState.valid)
+    setGlobalExpiration(expiration)
+  }, [amount, duration, setGlobalExpiration])
+
+  return (
+    <ModalWrapper>
+      <InputWrapper isInvalid={errorState !== ErrorState.valid}>
+        <NumericInput
+          as="input"
+          pattern="[0-9]"
+          borderStyle="none"
+          className={bodySmall}
+          color={{ placeholder: 'textSecondary', default: 'textPrimary' }}
+          value={amount}
+          width="32"
+          marginRight="4"
+          backgroundColor="none"
+          onChange={setCustomExpiration}
+          flexShrink="0"
+        />
+        <DropdownWrapper className={buttonTextMedium}>
+          <SortDropdown
+            dropDownOptions={durationOptions}
+            mini
+            miniPrompt={displayDuration + (displayDuration === duration ? pluralize(amount) : 's')}
+            left={38}
+            top={38}
+          />
+        </DropdownWrapper>
+      </InputWrapper>
+      {errorState !== ErrorState.valid && (
+        <ErrorMessage className={caption}>
+          {' '}
+          <WarningIcon /> {errorState === ErrorState.overMax ? 'Maximum 6 months' : 'Set duration'}
```

Comment:
nit: moving to `getMessageForError` + a switch/case for future extendability? (if any)

Reply:
We don't have any plans for future error messages for this dropdown